tfx_bsl/beam/run_inference.py

```diff
@@ -221,17 +233,213 @@ def update(self, elements: List[Union[tf.train.Example,
       self._inference_request_batch_byte_size.update(
           sum(element.ByteSize() for element in elements))
 
+  def __init__(self):
+    super(_BaseDoFn, self).__init__()
+    self._clock = None
+    self._metrics_collector = self._MetricsCollector()
+
+  def setup(self):
+    self._clock = _ClockFactory.make_clock()
+
+  def finish_bundle(self):
+    self._metrics_collector.update_metrics_with_cache()
+
+  def process(
+      self, elements: List[Union[tf.train.Example, tf.train.SequenceExample]]
+  ) -> Iterable[Any]:
+    batch_start_time = self._clock.get_current_time_in_microseconds()
+    outputs = self.run_inference(elements)
+    result = self._post_process(elements, outputs)
+    self._metrics_collector.update(
+        elements,
+        self._clock.get_current_time_in_microseconds() - batch_start_time)
+    return result
+
+  def run_inference(
+      self, elements: List[Union[tf.train.Example, tf.train.SequenceExample]]
+  ) -> Union[Mapping[Text, np.ndarray], Sequence[Mapping[Text, Any]]]:
+    raise NotImplementedError
+
+  def _post_process(
+      self, elements: List[Union[tf.train.Example, tf.train.SequenceExample]],
+      outputs: Any
+  ) -> Iterable[Any]:
+    raise NotImplementedError
+
+
+@beam.typehints.with_input_types(List[Union[tf.train.Example,
+                                            tf.train.SequenceExample]])
+# Using output typehints triggers NotImplementedError('BEAM-2717)' on streaming
```

Comment:
You can also add the commented-out typehint line that should be added (uncommented).